pkg/apis/seedmanagement/v1alpha1/types_managedseedset.go

```diff
@@ -0,0 +1,143 @@
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Spec defines the desired identities of ManagedSeeds and Shoots in this set.
+	// +optional
+	Spec ManagedSeedSetSpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
+	// Status is the current status of ManagedSeeds and Shoots in this ManagedSeedSet.
+	// +optional
+	Status ManagedSeedSetStatus `json:"status,omitempty" protobuf:"bytes,3,opt,name=status"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ManagedSeedSetList is a list of ManagedSeed objects.
+type ManagedSeedSetList struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard list object metadata.
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Items is the list of ManagedSeeds.
+	Items []ManagedSeedSet `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// ManagedSeedSetSpec is the specification of a ManagedSeedSet.
+type ManagedSeedSetSpec struct {
+	// Replicas is the desired number of replicas of the given Template. Defaults to 1.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,1,opt,name=replicas"`
+	// Selector is a label query over ManagedSeeds and Shoots that should match the replica count.
+	// It must match the ManagedSeeds and Shoots template's labels.
+	// +optional
+	Selector *metav1.LabelSelector `json:"selector" protobuf:"bytes,2,opt,name=selector"`
+	// Template describes the ManagedSeed that will be created if insufficient replicas are detected.
+	// Each ManagedSeed created / updated by the ManagedSeedSet will fulfill this template.
+	Template ManagedSeedTemplate `json:"template" protobuf:"bytes,3,opt,name=template"`
+	// ShootTemplate describes the Shoot that will be created if insufficient replicas are detected for hosting the corresponding ManagedSeed.
+	// Each Shoot created / updated by the ManagedSeedSet will fulfill this template.
+	ShootTemplate gardencorev1beta1.ShootTemplate `json:"shootTemplate,omitempty" protobuf:"bytes,4,rep,name=shootTemplate"`
+	// UpdateStrategy specifies the ManagedSeedSetUpdateStrategy that will be
+	// employed to update ManagedSeeds / Shoots in the ManagedSeedSet when a revision is made to
+	// Template / ShootTemplate.
+	// +optional
+	UpdateStrategy *ManagedSeedSetUpdateStrategy `json:"updateStrategy,omitempty" protobuf:"bytes,5,opt,name=updateStrategy"`
+	// RevisionHistoryLimit is the maximum number of revisions that will
+	// be maintained in the ManagedSeedSet's revision history. Defaults to 10.
+	// +optional
+	RevisionHistoryLimit *int32 `json:"revisionHistoryLimit,omitempty" protobuf:"varint,6,opt,name=revisionHistoryLimit"`
+}
+
+// ManagedSeedSetUpdateStrategy specifies the strategy that the ManagedSeedSet
+// controller will use to perform updates. It includes any additional parameters
+// necessary to perform the update for the indicated strategy.
+type ManagedSeedSetUpdateStrategy struct {
```

Comment:
Regarding the naming of all the types and consts related to the update strategy:
I think it would be fine to remove `ManagedSeedSet` from their names.
For example: `RollingUpdateManagedSeedSetUpdateStrategyType` is quite lengthy/bumpy and will be harder comprehend in code. We could just name it `RollingUpdateStrategyType` instead, as I aspect there won't be any other `Set` type/controller in this API group anytime soon and it's rather clear that the update strategy types/consts belong to the `ManagedSeedSet`.
WDYT?

Reply:
Yes, I agree, will shorten the names.